src/index.mjs

```diff
@@ -63,40 +43,89 @@ function isIgnored(node, filter) {
  * links for `document`. Can also work off a supplied
  * DOM element or static array of URLs.
  * @param {Object} options - Configuration options for quicklink
- * @param {Array} options.urls - Array of URLs to prefetch (override)
- * @param {Object} options.el - DOM element to prefetch in-viewport links of
- * @param {Boolean} options.priority - Attempt higher priority fetch (low or high)
- * @param {Array} options.origins - Allowed origins to prefetch (empty allows all)
- * @param {Array|RegExp|Function} options.ignores - Custom filter(s) that run after origin checks
- * @param {Number} options.timeout - Timeout after which prefetching will occur
- * @param {Function} options.timeoutFn - Custom timeout function
+ * @param {Object} [options.el] - DOM element to prefetch in-viewport links of
+ * @param {Boolean} [options.priority] - Attempt higher priority fetch (low or high)
+ * @param {Array} [options.origins] - Allowed origins to prefetch (empty allows all)
+ * @param {Array|RegExp|Function} [options.ignores] - Custom filter(s) that run after origin checks
+ * @param {Number} [options.timeout] - Timeout after which prefetching will occur
+ * @param {Number} [options.throttle] - The concurrency limit for prefetching
+ * @param {Number} [options.limit] - The total number of prefetches to allow
+ * @param {Function} [options.timeoutFn] - Custom timeout function
  */
-export default function (options) {
+export function listen(options) {
   if (!options) options = {};
+  if (!window.IntersectionObserver) return;
+
+  const [toAdd, isDone] = throttle(options.throttle || 1/0);
+  const limit = options.limit || 1/0;
 
-  observer && (observer.priority = options.priority || false);
+  // TODO: I think this isn't needed?
+  const isPriority = !!options.priority;
```

Comment:
I'm sure I just don't fully realize the purpose, but I don't think this is strictly necessary since anything that comes through `listen` passes thru the IO, which will only call `prefetch` once a link is visible. 

The only use I can see is when a user slices up their page(s) into smaller `options.el` containers, saying that ContainerA is high priority and the others aren't. Maybe that's a big one & makes it worth keeping 😆

Reply:
> The only use I can see is when a user slices up their page(s) into smaller options.el containers, saying that ContainerA is high priority and the others aren't. 

I haven't (to date) observed users heavily relying on granular `options.el` containers. I'm open to us considering this not-necessary and evaluating if we need to circle back during an alpha release. SG?

Reply:
Are you suggesting we drop `options.el` too? Or are you saying you're also on the fence about `options.priority`'s relevance? Sorry 🙈

Reply:
No worries!

After digging in further with a few users, it looks like folks _do_ need the ability to limit observation to a set of elements (e.g just navigation UI or just DOM elements corresponding to article content but not nav). That said.. I'm definitely on the fence about `options.priority`.

When it was first implemented, I thought there was a lot of value in providing the ability to fetch next pages/next resources with a high priority if it was properly scheduled. However, in retrospect, I don't believe this is actually necessary. If we just supported rel=prefetch and fallbacks without this notion of `priority` that would still serve the 90% use-case and also provide a way to further trim down bundle size.

Wdyt?

Reply:
We'd only be saving 9B by not having this option inside `listen` method – size isn't my driver here. I'm just not sure it has benefit in an IO context.

But, if we're certainly keeping `options.el`, then there _is_ at least one use case for this.

```
listen({
  el: document.getElementById('nav'),
  priority: true
})

listen({
  el: document.getElementById('wrapper'),
  // regular priority
  throttle: 3
})
```

I'm fine keeping it as is – someone somewhere will want to do this

Reply:
Keeping this as is sounds reasonable to me